### ISSUE TITLE
Call to a member function prepare() on null

### ISSUE THREAD

**Issue by idris11** [jenssegers/laravel-mongodb#1526]
I cant generate php artisan passport:install, I had try to change 

`<?php

namespace App;

use Illuminate\Notifications\Notifiable;
use Illuminate\Foundation\Auth\User as Authenticatable;

class User extends Authenticatable
{
    use Notifiable;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'name', 'email', 'password',
    ];

    /**
     * The attributes that should be hidden for arrays.
     *
     * @var array
     */
    protected $hidden = [
        'password', 'remember_token',
    ];
}'

into 
`<?php

namespace App;

use Illuminate\Notifications\Notifiable;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Jenssegers\Mongodb\Eloquent\Model as Eloquent;

class User extends Eloquent implements Authenticatable
{
    use Notifiable;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'name', 'email', 'password',
    ];

    /**
     * The attributes that should be hidden for arrays.
     *
     * @var array
     */
    protected $hidden = [
        'password', 'remember_token',
    ];
}`

but , it didnt work properly.
need help pls

**Comment by kwame101** [comment]
I'm not sure about your error maybe you can show the error log to see whats wrong with it but as I can see you are using auth interface so ;
Use this `use Jenssegers\Mongodb\Auth\User as Authenticatable` instead of  `use Illuminate\Foundation\Auth\User as Authenticatable`

**Comment by sempixel** [comment]
I guess it's linked to `MorphMany` missing functionality.

**Comment by olimarferraz** [comment]
Works for me @kwame101 , thanks!

**Comment by 123dipesh** [comment]
I got the same error while working with laravel cashier braintree and mongodb as database.

I solved it by editing in vendor
- go in vendor/laravel/cashier-braintree/src/subcription.php
- change  "use Illuminate\Database\Eloquent\Model"  line with  "use  Jenssegers\Mongodb\Eloquent\Model"
-Done

**Comment by blueneel05** [comment]
TRY THIS : https://github.com/sadnub/laravel-mongodb-passport-fix WORKED FOR ME

**Comment by devenpateldp** [comment]
Laravel Version: 5.7
PHP Version:7.1
Laravel-admin: latest

After first installing
https://github.com/jenssegers/laravel-mongodb

Symfony \ Component \ Debug \ Exception \ FatalThrowableError (E_ERROR)
Call to a member function prepare() on null

Symfony\Component\Debug\Exception\FatalThrowableError
…/vendor/laravel/framework/src/Illuminate/Database/Connection.php326

vendor/laravel/framework/src/Illuminate/Database/Connection.php

any solution?

**Comment by devenpateldp** [comment]
Laravel Version: 5.7
PHP Version:7.1
Laravel-admin: latest

After first installing
https://github.com/jenssegers/laravel-mongodb

Symfony \ Component \ Debug \ Exception \ FatalThrowableError (E_ERROR)
Call to a member function prepare() on null

Symfony\Component\Debug\Exception\FatalThrowableError
…/vendor/laravel/framework/src/Illuminate/Database/Connection.php326

vendor/laravel/framework/src/Illuminate/Database/Connection.php

any solution?

**Comment by nbyloff** [comment]
Yes this just started happening to me today on a query that's been running fine for over a year.

**Comment by Jchegenye** [comment]
@devenpateldp This should fix your issue.

> TRY THIS : https://github.com/sadnub/laravel-mongodb-passport-fix WORKED FOR ME

**Comment by devenpateldp** [comment]
@Jchegenye Thank you so much in advance let me try again.

**Comment by devenpateldp** [comment]
@Jchegenye tried same again still issue same. if anything else please let me know.


Symfony \ Component \ Debug \ Exception \ FatalThrowableError (E_ERROR)
Call to a member function prepare() on null

Application frames (1) All frames (2)
1
Symfony
\Component
\Debug
\Exception
\FatalThrowableError
…
/vendor
/laravel
/framework
/src
/Illuminate
/Database
/Connection.php
326
0
{main}
…
/public
/index.php
0

**Comment by sedrak1987** [comment]
use my package for mongodb(based to Laravel passport, so you can use Passport doc)
https://packagist.org/packages/sedrak/laravel-mongodb-passport
remove laarvel passport from composer and install this.

**Comment by devenpateldp** [comment]
@sedrak1987 thank you will check and report.

**Comment by Racsodia** [comment]
I recommend this library https://packagist.org/packages/narwy/laravel-mongodb-passport

**Comment by devenpateldp** [comment]
@Racsodia thank you let me review.

**Comment by AbouOpenSource** [comment]
I have also the same problem.
To fix it, i changed in all Passport file 
"use Illuminate\Database\Eloquent\Model" by 
"use Jenssegers\Mongodb\Eloquent\Model"

list of the files where i changed : /vendor/laravel/passport/src/AuthCode.php
                                                    /vendor/laravel/passport/src/Client.php
                                                    /vendor/laravel/passport/src/PersonalAccessClient.php
                                                    /vendor/laravel/passport/src/RefreshToken.php
                                                   /vendor/laravel/passport/src/Token.php
In my case, i wanted to use Passport with mongodb like database.

**Comment by hanzoelfantasma** [comment]
I have also the same problem, thank you to @AbouOpenSource he gave me the idea of search "use Illuminate\Database\Eloquent\Model" and changed in this file: /vendor/laravel/framework/src/Illuminate/Notifications/DatabaseNotification.

My problem was with notify method.
